### PR TITLE
Query count test

### PR DESCRIPTION
**Pull request by SchrodingersGat**
Run query counting on *all* API requests made during unit testing.

Should help to quickly reveal any inefficient API endpoints

### CODE REVIEW

**Review by matmair — APPROVED** [review]
LGTM

**Comment by matmair** [comment]
Looks like there are a lot of queries that trigger the 75 requests limit

File: src/backend/requirements.in [hunk 1]
```diff
@@ -18,6 +18,7 @@ django-maintenance-mode                 # Shut down application while reloading
 django-markdownify                      # Markdown rendering
 django-mptt                             # Modified Preorder Tree Traversal
 django-markdownify                      # Markdown rendering
+django-middleware-global-request        # Global request middleware
```
Comment (matmair) [inline src/backend/requirements.in:21]: Are you sure about this? This bleeds possibly sensitive request info into the python thread. Not sure how to shield this from plugins

Comment (SchrodingersGat) [reply src/backend/requirements.in:21]: I'm not sure about it - it feels like a pretty big hammer for the problem.

However caching this info against the request saves a heap of DB hits..

Actually thinking about it some more, the *better* way is to get a global cache working. The current plugin registry hash is coded around the fact that the default Django cache does not share between processes. If we had a global cache (redis or memcached) we could bypass this entirely.

The global request approach showed that there is a potential speed boost to be had, but I agree it's not the best approach.

File: src/backend/InvenTree/plugin/templatetags/plugin_extras.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 from common.models import InvenTreeSetting
 from common.notifications import storage
-from plugin import registry
+from plugin.registry import registry
```
Comment (matmair) [inline src/backend/InvenTree/plugin/templatetags/plugin_extras.py:9]: Question: Why this change? The before seems to be no different than the after as we are exposing registry on the plugin module level

**Review by matmair — APPROVED** [review]

File: src/backend/InvenTree/plugin/templatetags/plugin_extras.py (hunk 2 above)
Comment (SchrodingersGat) [reply src/backend/InvenTree/plugin/templatetags/plugin_extras.py:9]: I think the vscode IDE did not correctly introspect in the first case. So it was just a quality of life fix. Do you want me to revert it?

Comment (matmair) [reply src/backend/InvenTree/plugin/templatetags/plugin_extras.py:9]: I don’t particularly care either way, just seemed like a nonsensical change
